File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> map=new HashMap<>();
        for(int i=0;i<nums.length;i++){
            map.put(nums[i],i);
        }
        for(int i=0;i<nums.length;i++){
            int diff=target-nums[i];
            if(map.containsKey(diff)){
                if(map.get(diff)!=i){
                    int[] ans={i,map.get(diff)};
                    return ans;
                }
            }

        }
        return new int[2];
    }
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int i,j;
        int k=0;
        vector<int>a;
       for(i=0;i<nums.size()&&k==0;i++){
           for(j=i+1;j<nums.size();j++){
               if(nums[i]+nums[j]==target){
                   k=1;
                   break;
               }
           }
       } 
       a.push_back(i-1);
       a.push_back(j);
       return a;
    }
};
